src/pages/MainPage/components/FavoriteMovieBlock/index.tsx

```diff
@@ -1,63 +1,64 @@
 import MovieCard from "@components/MovieCard";
-import { ViewModeType, FavoriteMovie } from "types";
+import { ViewModeType } from "types";
 import { GRID_VIEW, LIST_VIEW, START_PAGE } from "@components/constants";
-import getPaginatedFavoriteMovies from "@services/getFavoriteMoviesPage";
 import Pagination from "@components/Pagination";
 import FavoriteMovieCardButtons from "./components/FavoriteMovieCardButtons";
+import useGetFavoriteMovies from "@gqlHooks/useGetFavoriteMovies";
+import { MOVIES_PER_PAGE } from "@components/constants";
+import { useToggleWatchedStatus } from "@gqlHooks/useToggleWatchedStatus";
+import { useDeleteFavoriteMovie } from "@gqlHooks/useDeleteFavoriteMovie";
 
 interface FavoriteMoviesBlockProps {
   genres: Genre[];
 }
 
 function FavoriteMoviesBlock({ genres }: FavoriteMoviesBlockProps) {
   const { t } = useLingui();
+  const navigate = useNavigate();
   const [viewMode, setViewMode] = useState<ViewModeType>(LIST_VIEW);
   const [currentPage, setPage] = useState(START_PAGE);
-  const [moviesOnPage, setMoviesOnPage] = useState<FavoriteMovie[]>([]);
-  const [totalPages, setTotalPages] = useState(0);
-  const navigate = useNavigate();
+  const { moviesOnPage, totalPages, loading, error } = useGetFavoriteMovies(
+    currentPage,
+    MOVIES_PER_PAGE
+  );
+  const toggleWatched = useToggleWatchedStatus(currentPage);
+  const deleteFavMovie = useDeleteFavoriteMovie(currentPage);
+  const toggleWatchedStatus = (id: number) => {
+    toggleWatched(id);
+  };
+  const handleDelete = (id: number) => {
+    deleteFavMovie(id);
+    if (moviesOnPage.length === 1 && currentPage !== START_PAGE) {
+      deleteFavMovie(id);
+      setPage(currentPage - 1);
+    }
+    deleteFavMovie(id);
+  };
   const handleAdd = () => {
     navigate("/searchmovies");
   };
 
-  const [favoriteMovies, setFavoriteMovies] = useState<FavoriteMovie[]>(
-    JSON.parse(localStorage.getItem("favoriteMovies") || "[]")
-  );
-
   useEffect(() => {
-    const { paginatedFavoriteMovies, totalPages } =
-      getPaginatedFavoriteMovies(currentPage);
-    setMoviesOnPage(paginatedFavoriteMovies);
-    setTotalPages(totalPages);
-  }, [currentPage, favoriteMovies]);
-
-  const toggleWatchedStatus = (id: number) => {
-    const updatedMovies = favoriteMovies.map((movie) => {
-      if (movie.id === id) {
-        return {
-          ...movie,
-          watchedStatus: !movie.watchedStatus,
-        };
-      }
-      return movie;
-    });
-    setFavoriteMovies(updatedMovies);
-    localStorage.setItem("favoriteMovies", JSON.stringify(updatedMovies));
-  };
+    if (!loading && moviesOnPage?.length === 0 && currentPage > START_PAGE) {
+      setPage((prevPage) => prevPage - 1);
+    }
+  }, [moviesOnPage, loading, currentPage]);
 
-  const handleDelete = (id: number) => {
-    const updatedMovies = favoriteMovies.filter(
-      (favMovie: FavoriteMovie) => favMovie.id !== id
+  if (loading) return <p>Loading...</p>;
+  if (totalPages == 0)
```

Comment:
используй только строгие сравнения
```suggestion
  if (totalPages === 0)
```

а если ошибка, totalPages тоже равно 0?

Reply:
Не, если ошибка, то единичка, ноль только в случае если ни один фильм не добавлен в избранные
в кастомном хуке useGetFavoriteMovies проверяю и задаю значение:
```
  const totalPages = data?.getFavoriteMovies.totalPages ?? 1;
```